plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  PluginDatabaseManager,
+  resolvePackagePath,
+} from '@backstage/backend-common';
+import {
+  NotificationGetOptions,
+  NotificationModifyOptions,
+  NotificationsStore,
+} from './NotificationsStore';
+import { Notification } from '@backstage/plugin-notifications-common';
+import { Knex } from 'knex';
+
+const migrationsDir = resolvePackagePath(
+  '@backstage/plugin-notifications-backend',
+  'migrations',
+);
+
+/** @public */
```

Comment:
Lot of things marked public here that aren't. Use `@internal` instead

Reply:
Yeah, I change these. Probably the `NotificationsStore` interface should be public still?

Reply:
Yep we'd add an extension point for that in the future if it's something we feel makes sense. We've generally backtracked on the idea of public storage interfaces though and usually avoid it